internal/handlers/pg/pgdb/query.go

```diff
@@ -147,37 +148,81 @@ func Explain(ctx context.Context, tx pgx.Tx, sp SQLParam) (*types.Array, error)
 
 // buildQuery builds SELECT or EXPLAIN SELECT query.
 //
-// It returns (possibly wrapped) ErrSchemaNotExist or ErrTableNotExist
-// if schema/database or table/collection does not exist.
-func buildQuery(ctx context.Context, tx pgx.Tx, sp *SQLParam) (string, error) {
+// It returns the query string and the arguments.
+// If schema/database or table/collection does not exist,
+// it returns (possibly wrapped) ErrSchemaNotExist or ErrTableNotExist.
+func buildQuery(ctx context.Context, tx pgx.Tx, sp *SQLParam) (string, []any, error) {
 	exists, err := CollectionExists(ctx, tx, sp.DB, sp.Collection)
 	if err != nil {
-		return "", lazyerrors.Error(err)
+		return "", nil, lazyerrors.Error(err)
 	}
+
 	if !exists {
-		return "", lazyerrors.Error(ErrTableNotExist)
+		return "", nil, lazyerrors.Error(ErrTableNotExist)
 	}
 
 	table, err := getTableName(ctx, tx, sp.DB, sp.Collection)
 	if err != nil {
-		return "", lazyerrors.Error(err)
+		return "", nil, lazyerrors.Error(err)
+	}
+
+	var query string
+
+	if sp.Explain {
+		query = `EXPLAIN (VERBOSE true, FORMAT JSON) `
 	}
 
-	q := `SELECT _jsonb `
+	query += ` SELECT _jsonb `
+
 	if c := sp.Comment; c != "" {
 		// prevent SQL injections
 		c = strings.ReplaceAll(c, "/*", "/ *")
 		c = strings.ReplaceAll(c, "*/", "* /")
 
-		q += `/* ` + c + ` */ `
+		query += `/* ` + c + ` */ `
 	}
-	q += `FROM ` + pgx.Identifier{sp.DB, table}.Sanitize()
 
-	if sp.Explain {
-		q = "EXPLAIN (VERBOSE true, FORMAT JSON) " + q
+	query += ` FROM ` + pgx.Identifier{sp.DB, table}.Sanitize()
+
+	var args []any
+
+	if sp.Filters != nil {
+		var where string
+
+		where, args = prepareWhereClause(sp.Filters)
+		query += where
+	}
+
+	return query, args, nil
+}
+
+// prepareWhereClause adds WHERE clause with given filters to the query and returns the query and arguments.
+func prepareWhereClause(sqlFilters *types.Document) (string, []any) {
+	var filters []string
+	var args []any
+	var p Placeholder
+
+	for k, v := range sqlFilters.Map() {
+		switch k {
+		case "_id":
+			switch v := v.(type) {
+			case *types.ObjectID:
+				filters = append(filters, fmt.Sprintf(`(_jsonb->'_id')::jsonb->>'$o' = %s`, p.Next()))
+
+				args = append(args, pgx.Identifier{string(must.NotFail(fjson.Marshal(v)))}.Sanitize())
+			}
+		default:
+			continue
+		}
+	}
+
+	var query string
+
+	if len(filters) > 0 {
+		query = ` WHERE ` + strings.Join(filters, " AND ")
```

Comment:
Let's add `(` and `)` so conditions would look like `WHERE (_id = 1) AND (_id = 2)`

Reply:
I added parentheses to filter generation part.